webviews/components/dropdown.tsx

```diff
@@ -0,0 +1,118 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as React from 'react';
+import { useState } from 'react'
+import { chevronIcon } from './icon';
+
+const enum KEYCODES {
+	esc = 27,
+	down = 40,
+	up = 38
+}
+
+export const Dropdown = ({ options, defaultOption, submitAction }) => {
+	const [selectedMethod, selectMethod] = useState<string>(defaultOption);
+	const [areOptionsVisible, setOptionsVisible] = useState<boolean>(false);
+
+	const EXPAND_OPTIONS_BUTTON = 'expandOptions';
+
+	const onClick = e => {
+		setOptionsVisible(!areOptionsVisible);
+	}
+
+	const onMethodChange = e => {
+		selectMethod(e.target.value);
+		setOptionsVisible(false);
+		const primaryButton = document.getElementById('confirm-button');
+		primaryButton.focus();
+	}
+
+	const onKeyDown = e => {
+		if (e.keyCode === KEYCODES.esc && areOptionsVisible) {
```

Comment:
Since this series of `if`s are mutually exclusive, it's slightly easier to read to make them `else if`s.

Reply:
I see now that this a copy-paste. 🤷‍♀️

Reply:
Restructured this to a `switch`, which I hope is easier to read